Sources/OpenAPIRuntime/Interface/HTTPBody.swift

```diff
@@ -0,0 +1,754 @@
+    @inlinable public convenience init(_ bytes: [UInt8]) {
+        self.init(bytes[...])
+    }
+}
+
+extension HTTPBody: ExpressibleByArrayLiteral {
+    public typealias ArrayLiteralElement = UInt8
+    public convenience init(arrayLiteral elements: UInt8...) {
+        self.init(elements)
+    }
+}
+
+extension HTTPBody {
+
+    /// Creates a new body from the provided data chunk.
+    /// - Parameter data: A single data chunk.
+    public convenience init(data: Data) {
+        self.init(ArraySlice(data))
+    }
+}
+
+extension Data {
+    /// Creates a string by accumulating the full body in-memory into a single buffer up to
+    /// the provided maximum number of bytes and converting it to `Data`.
+    /// - Parameters:
+    ///   - body: The HTTP body to collect.
+    ///   - maxBytes: The maximum number of bytes this method is allowed
+    ///     to accumulate in memory before it throws an error.
+    /// - Throws: `TooManyBytesError` if the body contains more
+    ///   than `maxBytes`.
+    public init(collecting body: HTTPBody, upTo maxBytes: Int) async throws {
+        self = try await Data(body.collect(upTo: maxBytes))
+    }
+}
+
+// MARK: - Underlying async sequences
+
+extension HTTPBody {
+
+    /// An async iterator of both input async sequences and of the body itself.
+    public struct Iterator: AsyncIteratorProtocol {
+
+        /// The element byte chunk type.
+        public typealias Element = HTTPBody.ByteChunk
+
+        /// The closure that produces the next element.
+        private let produceNext: () async throws -> Element?
+
+        /// Creates a new type-erased iterator from the provided iterator.
+        /// - Parameter iterator: The iterator to type-erase.
+        @usableFromInline init<Iterator: AsyncIteratorProtocol>(
+            _ iterator: Iterator
+        ) where Iterator.Element == Element {
+            var iterator = iterator
+            self.produceNext = {
+                try await iterator.next()
+            }
+        }
+
+        public mutating func next() async throws -> Element? {
+            try await produceNext()
+        }
+    }
+}
+
+extension HTTPBody {
+
+    /// A type-erased async sequence that wraps input sequences.
+    @usableFromInline struct BodySequence: AsyncSequence, Sendable {
+
+        /// The type of the type-erased iterator.
+        @usableFromInline typealias AsyncIterator = HTTPBody.Iterator
+
+        /// The byte chunk element type.
+        @usableFromInline typealias Element = ByteChunk
+
+        /// A closure that produces a new iterator.
+        @usableFromInline let produceIterator: @Sendable () -> AsyncIterator
+
+        /// Creates a new sequence.
+        /// - Parameter sequence: The input sequence to type-erase.
+        @inlinable init<Bytes: AsyncSequence>(_ sequence: Bytes) where Bytes.Element == Element, Bytes: Sendable {
+            self.produceIterator = {
+                .init(sequence.makeAsyncIterator())
```

Comment:
This one is initializing the local iterator (HTTPBody.Iterator), similar to https://github.com/apple/swift-openapi-runtime/pull/47#discussion_r1336069862.

Reply:
Sure, but the guidance we received was to avoid implicit memberwise init. That I made the wrong suggestion probably speaks to why.

Reply:
Agreed, these aren't great to read outside the IDE ☹️

I think we said to avoid specifically doing this `let foo: ExplicitType = .init(...)`, and use `let foo = ExplicitType(...)`. Using implicit inits is still used all over our codebase in other contexts, but happy to open that up for discussion as well (ideally, as long as we can have a swift-format rule for it 😄).